### PR TITLE
Add API for enabling smart contracts.

### LINKED ISSUE #3033: Add `casper_enable_contract_version` host function that enables previously disabled contracts in a package
Currently we only have a way to disable contracts in a package hash with `casper_disable_contract_version` and we need a functionality to enable contracts. 

This request's objective is;
1. to add `casper_enable_contract_version` host function that enables previously disabled contracts
2. Reword the comment on the corresponding `disable` ffi method to;
    - Disable the indicated contract version of the indicated contract package.
    - Disabled contract versions can no longer be executed (by any means)

### PR DESCRIPTION
**Pull request by mpapierski**
Closes #3033 

This PR adds a new API `storage::enable_contract_version` for smart contract developers, enabling a contract by its hash after disabling it with `storage::disable_contract_version.`

### CODE REVIEW

File: execution_engine/src/core/runtime/mod.rs [hunk 1]
```diff
@@ -1787,6 +1787,33 @@ where
...
+        contract_hash: ContractHash,
+    ) -> Result<Result<(), ApiError>, Error> {
+        let contract_package_key = contract_package_hash.into();
+        self.context.validate_key(&contract_package_key)?;
+
+        let mut contract_package: ContractPackage = self
+            .context
+            .get_validated_contract_package(contract_package_hash)?;
+
+        // Return an error in trying to disable the (singular) version of a locked contract.
```
Comment (Fraser999) [inline execution_engine/src/core/runtime/mod.rs]:
```suggestion
        // Return an error if trying to enable the (singular) version of a locked contract.
```
Although I'd be inclined to remove the comment altogether. I'm not sure what point it's trying to make - the "singular" seems important, but isn't really I think?

File: resources/local/chainspec.toml.in [hunk 2]
```diff
@@ -215,6 +215,7 @@ transfer_to_account = { cost = 2_500_000_000, arguments = [0, 0, 0, 0, 0, 0, 0]
 update_associated_key = { cost = 4_200, arguments = [0, 0, 0] }
 write = { cost = 14_000, arguments = [0, 0, 0, 980] }
 write_local = { cost = 9_500, arguments = [0, 1_800, 0, 520] }
+enable_contract_version = { cost = 200, arguments = [0, 0, 0, 0] }
```
Comment (Fraser999) [inline resources/local/chainspec.toml.in]: Aren't these ordered alphabetically currently?  I'd at least fix this one, the one in private and the one in production.

File: smart_contracts/contract/src/contract_api/storage.rs [hunk 3]
```diff
@@ -336,6 +336,30 @@ pub fn disable_contract_version(
     api_error::result_from(result)
 }
 
+/// Enable a version of a contract from the contract stored at the given
+/// `Key`. That version of the contract will no longer be callable by
```
Comment (Fraser999) [inline smart_contracts/contract/src/contract_api/storage.rs]: There's no `Key` here is there?

File: types/src/contracts.rs [hunk 4]
```diff
@@ -1813,6 +1825,50 @@ mod tests {
...
+                (next_version, NEW_CONTRACT_HASH),
+            ]),
+        );
+
+        assert_eq!(contract_package.disabled_versions(), &BTreeSet::new(),);
+
+        assert_eq!(
+            contract_package.current_contract_hash(),
+            Some(NEW_CONTRACT_HASH)
+        );
```
Comment (Fraser999) [inline types/src/contracts.rs:1877]: Should test that
* enabling an already-enabled contract returns Ok and is effectively a no-op
* enabling a non-existent version returns an error

File: smart_contracts/contract/src/contract_api/storage.rs [hunk 5]
```diff
@@ -336,6 +336,30 @@ pub fn disable_contract_version(
...
+/// `call_versioned_contract`. Note that this contract must have been created by
+/// `create_contract` or `create_contract_package_at_hash` first.
```
Comment (Fraser999) [inline smart_contracts/contract/src/contract_api/storage.rs]: `create_contract` is a private function.  I guess this should be:

```suggestion
/// [`new_contract`] or [`create_contract_package_at_hash`] first.
```

I think this change applies to a few other doc comments in this file.

**Review by Fraser999 — APPROVED** [review]
Nitpicks only from me.

**Comment by mpapierski** [comment]
Applied review comments, squashed and merged into #3106 PR in commit 45f661e302a8eb95b0108d1a08372c812b831b52

**Pull request closed without merging**